android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddleware.kt

```diff
@@ -146,40 +149,54 @@ class TranslationsMiddleware(
     }
 
     /**
-     * Checks if the translations engine supports the device architecture and updates the state.
-     * When the device does support the translations engine, then the engine will request
-     * initialization of translations data for the [BrowserState.translationEngine] via
-     * [TranslationsAction.InitTranslationsBrowserState].
+     * Use this to initialize translations data for [BrowserState.translationEngine]. If an
+     * issue occurs, the relevant error will be set on [BrowserState.translationEngine].
+     *
+     * This will populate:
+     * Language Support - [requestSupportedLanguages]
      *
      * @param context Context to use to dispatch to the store.
      */
-    private fun requestEngineSupportAndInit(
+    private fun initializeBrowserStore(
         context: MiddlewareContext<BrowserState, BrowserAction>,
     ) {
-        engine.isTranslationsEngineSupported(
-            onSuccess = { isEngineSupported ->
-                if (isEngineSupported) {
+        requestSupportedLanguages(context)
+    }
+
+    /**
+     * Checks if the translations engine supports the device architecture and updates the state on
+     * [BrowserState.translationEngine].
+     *
+     * @param context Context to use to dispatch to the store.
+     * @return Whether the engine is supported or not, or null when the support cannot be
+     * determined.
+     */
+    private suspend fun requestEngineSupport(
+        context: MiddlewareContext<BrowserState, BrowserAction>,
+    ): Boolean? {
+        return suspendCoroutine { continuation ->
```

Comment:
Is `suspendCoroutine` needed now?

Reply:
Wasn't sure the best or recommended way to structure this. Thinking was that I needed to wait for the value to resolve for `InitTranslationsBrowserState` to see if I can continue once the engine reports back. LMKWYT!

Reply:
That makes sense, just had to wrap my head around what was going on again. Thanks!